File: Data Structures & Algorithms/reverse-bits/submission-1.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t num = 0;
        for (int i = 0; i < 32; i++) {
            uint32_t bit = n & 1u;
            num |= bit;
            if (i != 31) num <<= 1u;
            n >>= 1u;
        }

        return num;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freqs = {}
        for num in nums:
            if num in freqs:
                freqs[num] += 1
            else:
                freqs[num] = 1
        
        buckets = [[] for x in range(len(nums) + 1)]
        for num in freqs:
            freq = freqs[num]
            buckets[freq].append(num)
        
        result = []
        for bucket in range(len(buckets)-1, 0, -1):
            for num in buckets[bucket]:
                result.append(num)
                if len(result) == k:
                    return result
        
        return result

        